shared/js/numerals_helper.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+(function(exports) {
+  var NumeralsHelper = {
+    _cnvEAWA: function(b) {
+      var allHTMLElements = [].slice
+        .call(document.querySelectorAll('[data-l10n-numerals]'));
+      allHTMLElements.forEach(function(el) {
+        var matches = el.textContent.match(b ? /\d+/g : /([٠-٩])/g);
```

Comment:
Can you use `String.prototype.replace` here and pass a function as the second arg?